iceoryx_posh/test/moduletests/test_roudi_service_registry.cpp

```diff
@@ -570,4 +570,42 @@ TEST_F(ServiceRegistry_test, SearchInFullRegistryWorks)
     ASSERT_EQ(searchResult.size(), 1);
 }
 
+using Entry = iox::roudi::ServiceRegistry::ServiceDescriptionEntry;
+
+TEST_F(ServiceRegistry_test, FunctionIsAppliedToAllEntriesInSearchResult)
```

Comment:
Please add here tests for the test case that the callable is not set. This can be done via the default ctor like this:
```cpp
cxx::function_ref<void(const ServiceContainer&)> searchFunction;
```